### PR TITLE
run wasm-pack build --target web on project

### PR DESCRIPTION
**Pull request by benschac**
After getting `wasm-pack` running on my machine locally, this diff is a result of running `wasm-pack build --target web`.

### Why

The current js build that is in the project targets nodejs, which will throw a null exception and crash the application. Targeting the web will give the project support for a web app, and fingers crossed a react native application.

 ### Reproduce
If you want to confirm changes in this diff run `wasm-pack build --target web`

### Assumptions
Assuming we don't need to support nodejs at the moment. I realize that this could possibly break tests in the SDK. If it's an issue I'll revisit this PR and add a nodejs js file.

### CODE REVIEW

**Comment by jakehemmerle** [comment]
great explanation of the PR, John and I were just discussing that we should have a basic template like this that every PR should include

Correct me if I'm wrong, but this PR seems to include the build (wasm binary), which is something rarely included in codebases. Am I wrong here or is there an explanation for this?

**Review by jakehemmerle — DISMISSED** [review]
Ah! I see John is the one who committed it to the repo, not you. lgtm!

File: example/test.ts [hunk 1]
```diff
@@ -1,5 +1,12 @@
...
-import {from_hex, to_hex, decrypt_and_verify, gen_signing_key, encrypt_and_sign, public_key_from_secret, constant_time_eq } from "x25519-chacha20poly1305";
+import {
+  to_hex,
+  from_hex,
+  gen_signing_key,
+  public_key_from_secret,
+  encrypt_and_sign,
+  decrypt_and_verify,
+  constant_time_eq,
+} from "../pkg/x25519_chacha20poly1305";
```
Comment (benschac) [inline example/test.ts:9]: not sure what changed here: I ran the same command that's in the makefile: 



`wasm-pack build --target web`

it's output is now in the `pkg` dir. There something I might have borked?

File: pkg/README.md [hunk 2]
```diff
@@ -2,12 +2,33 @@
 
 Nodejs package for x25519 key exchange and chacha20poly1305 encryption, written in Rust compiled to WASM.
 
+## Development process
```
Comment (benschac) [inline pkg/README.md:5]: this was autogenerated, can delete if not needed. Assumed I should just leave it for now.

File: pkg/x25519_chacha20poly1305.js [hunk 3]
```diff
@@ -361,6 +361,11 @@ module.exports.__wbindgen_is_string = function(arg0) {
     return ret;
 };
 
+module.exports.__wbg_msCrypto_6e7d3e1f92610cbb = function(arg0) {
```
Comment (benschac) [inline pkg/x25519_chacha20poly1305.js:364]: Assuming this was just wasm-pack moving something around? I don't think it's my listing / formatting. Just want to confirm with one of y'all. I'm still a bit new to rust/wasm tooling

File: example/test.ts (hunk 1 above)
Comment (jakehemmerle) [reply example/test.ts:9]: yours looks fine to me, just linting and pointing to the local build, which is what we want

**Comment by jakehemmerle** [comment]
lgtm, john should give final thumbs up

File: pkg/x25519_chacha20poly1305.js (hunk 3 above)
Comment (jakehemmerle) [reply pkg/x25519_chacha20poly1305.js:364]: yeah this is the build which again is unusual to have in the repo. I wouldn't worry much about the build files in `/pkg`

**Comment by ok-john** [comment]
> great explanation of the PR, John and I were just discussing that we should have a basic template like this that every PR should include

@jakehemmerle yes, perhaps we take this PR as a template?

> Correct me if I'm wrong, but this PR seems to include the build (wasm binary), which is something rarely included in codebases. Am I wrong here or is there an explanation for this?

We distribute the compiled wasm as part of a NPM library. As to the correctness of this, someone else should comment in. Perhaps @JesseAbram?

- - -

@benschac - we need to update the NPM package if this change is merged, can you look into doing that? Also can you add the NPM package link to the readme or somewhere else in this PR? Thanks @JesseAbram published the NPM package and can provide some insight there.

**Review by ok-john — CHANGES_REQUESTED** [review]
See latest comment.

**Comment by JesseAbram** [comment]
yes we do, I also think the best way to handle this issue going forward is to maintain 2 builds of the wasm, and then choose which one we use by knowing if we are in browser or not. like if window.object exists do web based if not do node based

**Comment by ok-john** [comment]
> yes we do, I also think the best way to handle this issue going forward is to maintain 2 builds of the wasm, and then choose which one we use by knowing if we are in browser or not. like if window.object exists do web based if not do node based

I've added an issue for this, could you sync with ben on updating the NPM package, perhaps we can have a github actions step do this upon any change to master so the process is automated?

**Comment by JesseAbram** [comment]
ya it shouldn't be hard the issue is how best to handle it on this repo, as in having two compile targets and pushing them both up to npm as different packages......or handling it internally within one package

**Comment by benschac** [comment]
> @jakehemmerle yes, perhaps we take this PR as a template?


Always down for templates, lists, and any way to bring down the cognitive load and help your reviewer.


> @benschac - we need to update the NPM package if this change is merged, can you look into doing that? Also can you add the NPM package link to the readme or somewhere else in this PR? Thanks @JesseAbram published the NPM package and can provide some insight there.

Yep! I have a plan. I think I have an issue but I'll double check right now.

I can make an issue with a bit more detail.

> Also can you add the NPM package link to the readme or somewhere else in this PR?

I'll add it in a follow up PR right now.

**Comment by benschac** [comment]
following up on this PR:

Will address these comments in:
- #7 , and close out this PR in favor of my new one that updates the wasm builds to support, node, web, and bundle js.

**Pull request closed without merging**